sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -275,36 +206,49 @@ public void populateDisplayData(DisplayData.Builder builder) {
             .add(DisplayData.item("inputMetadata", sb.toString())
             .withLabel("Input Metadata"));
       }
-      if (Boolean.TRUE.equals(getParseSynchronously())) {
-        builder
-          .add(DisplayData.item("parseMode", "synchronous")
-            .withLabel("Parse Mode"));
-      } else {
-        builder
-          .add(DisplayData.item("parseMode", "asynchronous")
-            .withLabel("Parse Mode"));
-        builder
-          .add(DisplayData.item("queuePollTime", getQueuePollTime().toString())
-            .withLabel("Queue Poll Time"))
-        .add(DisplayData.item("queueMaxPollTime", getQueueMaxPollTime().toString())
-          .withLabel("Queue Max Poll Time"));
-      }
-      Integer minTextLen = getMinimumTextLength();
-      if (minTextLen != null && minTextLen > 0) {
-        builder
-        .add(DisplayData.item("minTextLen", getMinimumTextLength().toString())
-          .withLabel("Minimum Text Length"));
-      }
-      if (Boolean.TRUE.equals(getReadOutputMetadata())) {
-        builder
-          .add(DisplayData.item("readOutputMetadata", "true")
-            .withLabel("Read Output Metadata"));
-      }
     }
 
-    @Override
-    protected Coder<String> getDefaultOutputCoder() {
-      return StringUtf8Coder.of();
+    private static class ParseToStringFn extends DoFn<ReadableFile, ParseResult> {
+
+      private static final long serialVersionUID = 6837207505313720989L;
+      private TikaIO.ParseAll spec;
+      ParseToStringFn(TikaIO.ParseAll spec) {
+        this.spec = spec;
+      }
+      @ProcessElement
+      public void processElement(ProcessContext c) throws IOException {
+        ReadableFile file = c.element();
+        InputStream stream = Channels.newInputStream(file.open());
+
+        final InputStream is = TikaInputStream.get(stream);
+        TikaConfig tikaConfig = null;
+        if (spec.getTikaConfigPath() != null) {
+          try {
+            tikaConfig = new TikaConfig(spec.getTikaConfigPath().get());
```

Comment:
This still needs to be fixed to use `FileSystems` API to read the config, because tikaConfigPath will not be accessible via regular java file APIs when the pipeline runs on anything but a local machine.

Reply:
Tried to, please check if that is correct...